From b1952d3d4cd2668b5140eb93ec83748dd80ab66d Mon Sep 17 00:00:00 2001
From: karol95c <292419@uwr.edu.pl>
Date: Tue, 23 Mar 2021 13:53:22 +0100
Subject: [PATCH] application_wpe start up in preloading state.

Fixes issue reported by MCA with blank screen observed on Cobalt activation.
Changed start up from Starting state to Preloading state to avoid initializing
resources and immediately deinitializing due to going straight to suspend mode.
Like in Starting state application is running but in Preloading state is not visible and receive no input.
---
 .../starboard/wpe/shared/application_wpe.cc            | 10 ++++++++++
 src/third_party/starboard/wpe/shared/application_wpe.h |  2 ++
 2 files changed, 12 insertions(+)

diff --git a/src/third_party/starboard/wpe/shared/application_wpe.cc b/src/third_party/starboard/wpe/shared/application_wpe.cc
index 8a5726918..b03c6f533 100644
--- a/src/third_party/starboard/wpe/shared/application_wpe.cc
+++ b/src/third_party/starboard/wpe/shared/application_wpe.cc
@@ -167,6 +167,16 @@ void Application::Inject(Event* e) {
   QueueApplication::Inject(e);
 }
 
+bool Application::IsStartImmediate()
+{
+  return false;
+}
+
+bool Application::IsPreloadImmediate()
+{
+  return true;
+}
+
 }  // namespace shared
 }  // namespace wpe
 }  // namespace starboard
diff --git a/src/third_party/starboard/wpe/shared/application_wpe.h b/src/third_party/starboard/wpe/shared/application_wpe.h
index 2f0735ae7..d82cc1483 100644
--- a/src/third_party/starboard/wpe/shared/application_wpe.h
+++ b/src/third_party/starboard/wpe/shared/application_wpe.h
@@ -57,6 +57,8 @@ class Application : public ::starboard::shared::starboard::QueueApplication {
   void OnSuspend() override;
   void OnResume() override;
   void Inject(Event* e) override;
+  bool IsStartImmediate() override;
+  bool IsPreloadImmediate() override;
 
   // --- QueueApplication overrides ---
   bool MayHaveSystemEvents() override;
